docs/guides/data-movement/copy-offload.md

```diff
@@ -0,0 +1,92 @@
+
+```console
+CLIENT_TLS_SECRET=nnf-dm-usercontainer-client-tls
+kubectl get secrets $CLIENT_TLS_SECRET -o json | jq -rM '.data."tls.crt"' | base64 -d > cert.pem
+```
+
+> [!IMPORTANT]
+> Copy the certificate to `/etc/nnf-dm-usercontainer/cert.pem` on each compute node. It must be readable by all users' compute applications.
+
+### Rabbit hostname in /etc/hosts
+
+Each compute node must know the name of its matching rabbit node. This allows the `libcopyoffload` library, used in the user's compute application, to know which rabbit has the copy-offload server for that job. The library expects to use a generic name, `local-rabbit`, to contact that rabbit.
+
+### WLM and the per-Workflow token
+
+The WLM, such as Flux, must retrieve the per-Workflow token and make it available to the user's compute application as an environment variable named `DW_WORKFLOW_TOKEN`. The token is used by the `libcopyoffload` library to construct the "Bearer Token" for its requests to the copy-offload server. The token becomes invalid after the Workflow enters its teardown state.
+
+The Workflow contains a reference to the name of the Secret that holds the token. The following value returns the name and namespace of the secret:
+
+```console
+kubectl get workflow $WORKFLOW_NAME -o json | jq -rM '.status.workflowToken'
+```
+
+If information about the token's secret is returned, then read the token from the given secret:
+
+```console
+TOKEN=$(kubectl get secret -n $SECRET_NAMESPACE $SECRET_NAME -o json | jq -rM '.data.token' | base64 -d)
+```
+
+Create the environment variable for the user's compute application:
+
+```bash
+DW_WORKFLOW_TOKEN="$TOKEN"
+```
+
+> [!NOTE]
+> Per-Workflow tokens are not limited to the copy-offload API. Any user container may request to be configured with the job's per-Workflow token and the TLS certificate. See `requires=user-container-auth` in [User Containers](../user-containers/readme.md). The WLM must always check for the existence of a token secret in the Workflow.
+
+## User Enablement of Copy Offload
+
+Users enable the copy-offload server by requesting it in their job script. The script must contain a `DW container` directive that specifies the desired copy-offload container profile. At least one of the `DW jobdw` or `DW persistentdw` directives in the job script must include the `requires=copy-offload` statement. See [User Interactions](../user-interactions/readme.md) for more details about these directives.
+
+The user's compute application must be linked with the `libcopyoffload` library. This library understands how to find and use the TLS certificate and the per-Workflow token required for communication with the copy-offload server for the user's job.
+
+The copy-offload server and its container profile are specified in the `container` directive. See [User Containers](../user-containers/readme.md) for details about container profiles. The following selects the default copy-offload container profile:
```

Comment:
It's really just the profile right? This reads like the user needs to specify the server itself.

Reply:
I see what you mean. I'll rewrite that.